features/schedule/basic/feature.ts

```diff
@@ -0,0 +1,112 @@
+import { Feature, retry } from '@temporalio/harness';
+import { ScheduleClient, Connection, ScheduleOverlapPolicy } from '@temporalio/client';
+import * as assert from 'assert';
+import { randomUUID } from 'crypto';
+
+export async function BasicWorkflow(arg: string): Promise<string> {
+  return arg;
+}
+
+export const feature = new Feature({
+  workflow: BasicWorkflow,
+  alternateRun: async (runner) => {
+    const connection = await Connection.connect({
+      address: runner.options.address,
+      tls: runner.options.tlsConfig,
+    });
+
+    const client = new ScheduleClient({
+      connection,
+      namespace: runner.options.namespace,
+    });
+    const scheduleId = `schedule-basic-${randomUUID()}`;
+    const workflowId = `schedule-basic-workflow-${randomUUID()}`;
+    const handle = await client.create({
+      scheduleId,
+      spec: {
+        intervals: [{ every: '2s' }],
+      },
+      action: {
+        type: 'startWorkflow',
+        workflowId,
+        workflowType: BasicWorkflow,
+        args: ['arg1'],
+        taskQueue: runner.options.taskQueue,
+      },
+      policies: {
+        overlap: ScheduleOverlapPolicy.BUFFER_ONE,
+      },
+    });
+
+    try {
+      // Test describing
+      const desc = await handle.describe();
+      assert.equal(desc.scheduleId, scheduleId);
+      assert.equal(desc.action.workflowId, workflowId);
+
+      // Test listing
+      // Advanced visibility is eventually consistent. See https://github.com/temporalio/sdk-features/issues/182
+      assert.ok(
+        await retry(async function () {
+          for await (const schedule of client.list({ pageSize: 1 })) {
+            if (schedule.scheduleId == scheduleId) {
+              return true;
+            }
+          }
+          return false;
+        }, 10)
+      );
+
+      const waitCompletedWith = async (untilResult: string) => {
+        const iterable = await runner.client.list({
+          query: "WorkflowType = 'BasicWorkflow'",
+        });
+        for await (const exec of iterable) {
+          if (!exec.workflowId.startsWith(workflowId)) {
+            continue;
+          }
```

Comment:
Is this additional filtering required due to limitations of "basic visibility"?
Can we add a TODO to fix the query once this is supported?

Reply:
I don't think it is a limitation of basic visibility? We want to match just on the beginning and since `WorkflowID` is a keyword it only supports full matching regardless of basic vs advanced visibility.

Reply:
OIC, I missed that we're using `startsWith`, is there another search attribute we can leverage?

Reply:
I'm facing a similar issue in another test. On most SQL backends, startwith x can be implemented using a Between x and xzzzzzz... but that idiom seems to fail in SQLite...

Reply:
I considered generating a random task queue name, but that attribute is not searchable in sql backends. Also considered to make the workflow type random, but that'a ridiculously complex to implement because we don't have catch all workflows handlers...

Reply:
Part of what this is testing is that the workflow ID has the correct form so there is no other search attribute that can help us do that.